depparse/src/main/java/edu/illinois/cs/cogcomp/depparse/DepAnnotator.java

```diff
@@ -0,0 +1,107 @@
+import java.net.URL;
+
+public class DepAnnotator extends Annotator {
+    private static Logger logger = LoggerFactory.getLogger(DepAnnotator.class);
+    private static SLModel model = null;
+    private static final String MODEL_NAME = "struct-perceptron-auto-20iter.model";
+    private static final String TEMP_MODEL_FILE_NAME = "tmp345673.model";
+
+    public DepAnnotator() {
+        super(ViewNames.DEPENDENCY, new String[] {ViewNames.POS, ViewNames.SHALLOW_PARSE,
+                ViewNames.LEMMA});
+    }
+
+    @Override
+    public void initialize(ResourceManager rm) {
+        try {
+            // TODO Ugly hack: SL doesn't accept streams and can't create a file from inside a jar
+            File dest = new File(TEMP_MODEL_FILE_NAME);
+            URL fileURL = IOUtils.lsResources(DepAnnotator.class, MODEL_NAME).get(0);
+            logger.info("Loading {} into temp file: {}", MODEL_NAME, TEMP_MODEL_FILE_NAME);
+            FileUtils.copyURLToFile(fileURL, dest);
+            model = SLModel.loadModel(TEMP_MODEL_FILE_NAME);
+            ((LabeledChuLiuEdmondsDecoder) model.infSolver).loadDepRelDict();
+            if (!dest.delete())
+                throw new IOException("Could not delete temporary model file "
+                        + TEMP_MODEL_FILE_NAME);
+        } catch (IOException | ClassNotFoundException | URISyntaxException e) {
+            e.printStackTrace();
+            File dest = new File(TEMP_MODEL_FILE_NAME);
+            if (!dest.delete())
+                throw new RuntimeException("Could not delete temporary model file "
+                        + TEMP_MODEL_FILE_NAME);
+        }
+    }
+
+    @Override
+    public void addView(TextAnnotation ta) throws AnnotatorException {
+        for (String reqView : requiredViews)
+            if (!ta.hasView(reqView))
+                throw new AnnotatorException("TextAnnotation must have view: " + reqView);
+
+        DepInst sent = new DepInst(ta);
+        DepStruct deptree;
+        try {
+            deptree = (DepStruct) model.infSolver.getBestStructure(model.wv, sent);
+        } catch (Exception e) {
+            throw new AnnotatorException("Sentence cannot be parsed");
+        }
+
+        TreeView treeView = new TreeView(ViewNames.DEPENDENCY, ta);
+        int rootPos = findRoot(deptree);
+        // All the node positions are -1 to account for the extra <root> node added
+        Pair<String, Integer> nodePair = new Pair<>(sent.forms[rootPos], rootPos - 1);
+        Tree<Pair<String, Integer>> tree = new Tree<>(nodePair);
+        populateChildren(tree, deptree, sent, rootPos);
+        treeView.setDependencyTree(0, tree);
+        ta.addView(ViewNames.DEPENDENCY, treeView);
+    }
+
+    private void populateChildren(Tree<Pair<String, Integer>> tree, DepStruct struct, DepInst sent,
+            int pos) {
+        for (int i : struct.deps.get(pos)) {
+            // All the node positions are -1 to account for the extra <root> node added
```

Comment:
Thank you for adding this comment. I imagine it will save approx. 2.7 programmer-years of collective CCG programmer time over the next decade.